### DISCUSSION TITLE
GitHub Action - Vercel Authorization Query

### DISCUSSION THREAD

**Discussion by amar9elbeg** [vercel/community#3397] [Feature Question] [answered]
### Summary

Dear Vercel Support Team,

I hope this message finds you well. We are currently using GitHub Actions to deploy numerous projects to Vercel. To streamline our workflows and avoid the need to pass the Vercel token in each command, we are looking for a way to authenticate or authorize Vercel within the GitHub Action once.

Is there a way to log in to Vercel or authorize it within a GitHub Action, so that subsequent Vercel commands can run without requiring the token explicitly? This would greatly simplify our workflow setup and enhance security by reducing token exposure in scripts.

If there's a specific way or best practice to achieve this, we would greatly appreciate your guidance or any documentation you can provide.

Thank you for your attention to this matter. We value your support and look forward to your response.

### Steps to Reproduce

no need to reproduce

**Comment by amyegan** [comment, marked as answer]
Hey @amar9elbeg. Storing the access token, org ID, and project ID as [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) is best practice. You can find a longer explanation of the recommended setup at https://vercel.com/guides/how-can-i-use-github-actions-with-vercel
